code/workspaces/web-app/src/components/stacks/StackCard.js

```diff
@@ -105,12 +111,18 @@ const StackCard = ({ classes, stack, openStack, deleteStack, editStack, restartS
   const expireStacks = getFeatureFlags().expireUnusedNotebooks;
 
   const generateWarningMessage = () => {
-    if (expireStacks && stack.accessTime) {
-      const { accessTimeWarning } = expireStacks;
+    if (expireStacks && stack.accessTime && stack.type) {
```

Comment:
Agree - I did wonder about that at the time, but decided to enforce the requirement for stack.type as it would simply not show any message if there is no stack.type.  You are right though, I was incorrectly trying to prevent a referenceError at line 119 where there is (stackTypes.RSTUDIO === stack.type) - but that would only throw a reference error if stack was undefined (which would have already happened) - the test will not throw a referenceError, it will correctly return false if 'type' doesn't exist.  I'll just remove the stack.type from line 114